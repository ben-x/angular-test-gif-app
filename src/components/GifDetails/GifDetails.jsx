import React from 'react';

import {GifPageDetails, GifPageimage, GifPageDiv} from './GifDetails.styles'

const GifDetails = ({data, clearGif}) => {
  return(
  <>
 <h2 onClick={clearGif} style={{ cursor:'pointer' }}> <i class="fa fa-arrow-left" aria-hidden="true"></i> &nbsp; Back</h2> 
<GifPageDiv>
  <GifPageimage src={data.images.original.url} />
  <GifPageDetails>
   <p> Title: <span> {data.title}</span></p>
   <p> Rating: <span>{data.rating}</span></p>
  </GifPageDetails>
</GifPageDiv>
  </>
)};

export default GifDetails;
