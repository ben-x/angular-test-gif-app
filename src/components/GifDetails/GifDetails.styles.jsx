import styled from 'styled-components';

const GifPageDiv = styled.div`
display: flex;
justify-content: center;
flex-direction: column
`;

const GifPageimage = styled.img`
margin-top:20px;
border-radius:5px;
width:100%
`;

const GifPageDetails = styled.div`
 padding: 10px;
 margin: 0 30%;

 >p {
   font-weight:bold;

   span {
   font-weight:normal;
   font-style: italic;
   }
 }

`;

export {GifPageDetails, GifPageimage, GifPageDiv}
