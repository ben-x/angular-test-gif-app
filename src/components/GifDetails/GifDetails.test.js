import React from "react";
import { shallow } from "enzyme";
import GifDetails from "./GifDetails";

const props = {
    data: {},
    clearGif: jest.fn()
};

const wrap = shallow(<GifDetails {...props} />);

describe("<GifDetails />", () => {
    it("renders the component successfully", () => {
        expect(wrap).toMatchSnapshot();
    });
});
