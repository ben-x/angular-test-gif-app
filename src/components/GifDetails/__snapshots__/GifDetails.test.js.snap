// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifDetails /> renders the component successfully 1`] = `ShallowWrapper {}`;
